crates/cache/src/config.rs

```diff
@@ -579,6 +586,187 @@ impl CacheConfig {
     }
 }
 
+/// A builder for `CacheConfig`s.
+#[derive(Debug, Clone, Default)]
+pub struct CacheConfigBuilder {
+    enabled: bool,
+    directory: Option<PathBuf>,
+    worker_event_queue_size: Option<u64>,
+    baseline_compression_level: Option<i32>,
+    optimized_compression_level: Option<i32>,
+    optimized_compression_usage_counter_threshold: Option<u64>,
+    cleanup_interval: Option<Duration>,
+    optimizing_compression_task_timeout: Option<Duration>,
+    allowed_clock_drift_for_files_from_future: Option<Duration>,
+    file_count_soft_limit: Option<u64>,
+    files_total_size_soft_limit: Option<u64>,
+    file_count_limit_percent_if_deleting: Option<u8>,
+    files_total_size_limit_percent_if_deleting: Option<u8>,
+}
+
+impl CacheConfigBuilder {
+    /// Specifies whether the cache system is used or not.
+    pub fn enabled(mut self, enabled: bool) -> Self {
```

Comment:
If you're ok I'd prefer to leave this not exposed, it's a bit weird to me to go through the rigamarole of creating a config just to say it's disabled. In that sense I think we should consider `CacheConfigBuilder` as "always create something with the `enabled` flag set to `true`".

Reply:
Agreed this indeed felt weird as well, I think we can go with an option on the main config and handle it that way